controller/pkg/secrets/cert.go

```diff
@@ -94,14 +94,36 @@ limitations under the License.
 
 // signCertificate takes a CertificateSigningRequest and a TLS Secret and returns a PEM x.509 certificate
 // signed by the certificate in the Secret.
-func signCertificate(csr certificatesv1.CertificateSigningRequest, ca *corev1.Secret) ([]byte, error) {
+func signCertificate(
+	csr certificatesv1.CertificateSigningRequest,
+	ca *corev1.Secret,
+	keyType mgrconfig.KeyType,
+) ([]byte, error) {
 	caKeyBlock, _ := pem.Decode(ca.Data["tls.key"])
 	if caKeyBlock == nil {
 		return nil, fmt.Errorf("failed decoding 'tls.key' data from secret %s", ca.Name)
 	}
-	priv, err := x509.ParseECPrivateKey(caKeyBlock.Bytes)
-	if err != nil {
-		return nil, err
+
+	var (
+		signatureAlgorithm x509.SignatureAlgorithm
+		priv               crypto.Signer
+		err                error
+	)
+	switch keyType {
+	case mgrconfig.ECDSA:
+		priv, err = x509.ParseECPrivateKey(caKeyBlock.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		signatureAlgorithm = x509.ECDSAWithSHA256
+	case mgrconfig.RSA:
+		priv, err = x509.ParsePKCS1PrivateKey(caKeyBlock.Bytes)
+		if err != nil {
+			return nil, err
+		}
+		signatureAlgorithm = x509.SHA256WithRSA
+	default:
+		return nil, fmt.Errorf("unsupported key type: %s", keyType)
 	}
```

Comment:
Nit: I think this could be cleaner if we had a separate function for it (e.g. `parsePrivateKey(key []byte) (crypto.Signer, x509.SignatureAlgorithm, error)`). WDYT?

Reply:
👍 96680d612301c2a52f6b2aaddc8cd9949595ecc8